docs/enhancements/infrastructure-inventory-services-api/resource-server.md

```diff
@@ -0,0 +1,372 @@
+  "ztp-done": ""
+}
+```
+
+For nodes from hardware managers:
+
+```json
+{
+  ...
+  TBD
+  ...
+}
+
+```
+#### Resource
+
+For nodes from ACM:
+
+```json
+{
+  "beta.kubernetes.io/arch": "amd64",
+  "beta.kubernetes.io/os": "linux",
+  "kubernetes.io/arch": "amd64",
+  "kubernetes.io/hostname": "hubcluster-ctlplane-1.example.com",
+  "kubernetes.io/os": "linux",
+  "node-role.kubernetes.io/control-plane": "",
+  "node-role.kubernetes.io/master": "",
+  "node-role.kubernetes.io/worker": "",
+  "node.openshift.io/os_id": "rhcos"
+}
+```
+
+For nodes from hardware managers:
+
+```json
+{
+  "admin-state": "UNLOCKED",
+  "operational-state": "ENABLED",
+  "tenant": "default_tenant",
+  "serial": "TBD",
+  "sockets": "TBD",
+  "cores": "TBD",
+  "bios": "TBD",
+  "memory": "TBD",
+  "cpu-model": "Intel(R) Xeon(R) Gold 5318Y CPU @ 2.10GHz"
+}
+
+```
+
+#### Deployment Manager
+
+```json
+{
+  "profileName": "k8s",
+  "profileData": {
+    "admin_client_cert": "... base64 public certificate ...",
+    "admin_client_key": "... base64 private key ...",
+    "admin_user": "system:admin",
+    "cluster_api_endpoint": "https://api.hubcluster.example.com:6443",
+    "cluster_ca_cert": "... base64 CA certificate ..."
+  }
+}
+```
+
+### Initialization
+
+The schema referenced above should be applied during deployment using a migration tool as specified
+[here](#tooling-and-general-dev-guidelines).
+
+### Data Synchronization
+
+Resource data can be acquired from multiple data sources (i.e., ACM, hardware managers, etc). Some data sources may
+support sending asynchronous notifications about changes to their data while other data sources may not support such
+mechanisms. For those that do not support such mechanisms, and possibly to protects against deficiencies with those
+that do, it is important to support a periodic audit/synchronization mechanism whereby we poll the data source for a
+snapshot of its current database contents. That snapshot data can be used to refresh our tables. To ensure that we
+are capable of removing stale data from our tables we employ a "generation id" value for each data source and for each
+piece of data that we store in our database. When a new snapshot is taken the generation id of the data source is
+incremented, and the generation id stored against each individual data tuple is aligned to that value. At the end of
+the synchronization phase for that data source any data tuple that has a generation id less than that of its data source
+is purged from the database.
+
+Snapshots should be taken each time the microservice is restarted and periodically every X minutes; where X should be
+adjusted to a reasonable value balancing the cost of re-synchronizing versus the cost of lack of up-to-date data.
+
+Ideally, since re-syncing is a resource intensive process, it would be best to not do a periodic sync for any data
+source which supports an asynchronous event notifications mechanism. However, given that these other software
```

Comment:
Implementation details...but if all these resources are k8s CRs, we can simply setup a watch on them with client-go? We basically setup a listen with `SharedInformerFactory` and just react on based on events (add/delete/etc). This way we dont really need to keep track of generation IDs and such (k8s will take care of that for us)

Reply:
No, they won't be k8s CRs in most cases because we'll be importing data from ACM, and h/w managers... so direct polling will be required.